deps/mbedtls.mk

```diff
@@ -23,6 +23,8 @@ $(SRCCACHE)/$(MBEDTLS_SRC)/source-extracted: $(SRCCACHE)/$(MBEDTLS_SRC).tar.gz
 	$(JLCHECKSUM) $<
 	mkdir -p $(dir $@) && \
 	$(TAR) -C $(dir $@) --strip-components 1 -xf $<
+	# Apply workaround for CMake 3.18.2 bug (https://github.com/ARMmbed/mbedtls/pull/3691).
+	cd $(SRCCACHE)/$(MBEDTLS_SRC) && patch -p1 -f < $(SRCDIR)/patches/mbedtls-cmake-findpy.patch
```

Comment:
Typically the way we do this is is to have each patch have a `.patch-applied` target file, like this
https://github.com/JuliaLang/julia/blob/master/deps/libgit2.mk#L38-L41

Then we build a chain of the patches depending on eachother, and finally have the `build-configured` file depend on those:
https://github.com/JuliaLang/julia/blob/master/deps/libgit2.mk#L49-L51

The prevents errors when you are switching between two versions of the source; the `source-extracted` target won't get re-run if the file already exists, so we don't want things like patches to get skipped if an older version of the source was already built.

Reply:
Cheers, thanks for the pointer! I will give adding this mechanism a whirl.

I mimicked the pcre build/patch for this pull request, which doesn't seem to employ this mechanism? Should the pcre build/patch be touched up accordingly?

Reply:
While reading through more of these makefiles, I noticed in mbedtls.mk#22-34 on this branch
```
$(SRCCACHE)/$(MBEDTLS_SRC)/source-extracted: $(SRCCACHE)/$(MBEDTLS_SRC).tar.gz
	$(JLCHECKSUM) $<
	mkdir -p $(dir $@) && \
	$(TAR) -C $(dir $@) --strip-components 1 -xf $<
	# Apply workaround for CMake 3.18.2 bug (https://github.com/ARMmbed/mbedtls/pull/3691).
	cd $(SRCCACHE)/$(MBEDTLS_SRC) && patch -p1 -f < $(SRCDIR)/patches/mbedtls-cmake-findpy.patch
	# Force-enable MD4
	sed "s|//#define MBEDTLS_MD4_C|#define MBEDTLS_MD4_C|" -i $(SRCCACHE)/$(MBEDTLS_SRC)/include/mbedtls/config.h
	touch -c $(SRCCACHE)/$(MBEDTLS_SRC)/CMakeLists.txt # old target
	echo 1 > $@

checksum-mbedtls: $(SRCCACHE)/$(MBEDTLS_SRC).tar.gz
	$(JLCHECKSUM) $<
```
I see that `$(JLCHECKSUM) $<` appears twice, once in the first line of the recipe for the `$(SRCCACHE)/$(MBEDTLS_SRC)/source-extracted` target and again in the recipe for the `checksum-mbedtls` target. Should the former instance go away and the former target depend upon the latter target? Or the latter target go away? Or something else? :)

Reply:
I've (hopefully) implemented the mechanism you outlined above. Let me know if it needs any touchups :).

Reply:
Regarding the checksum repetition mentioned above, I see that repetition also occurs in other makefiles, so I'm guessing there's a good reason for it? :)

Reply:
The issue is that `checksum-*` is a phony target and if we have non-phony targets (e.g. ones that create files as output and are not re-run if those files exist) depend on phony ones, the non-phony ones always get re-run.  So it's better for us to repeat ourselves a little bit to get a target that can be skipped if `source-extracted` already exists, and also a target that will always run no matter what.

This is important because we use the `checksum-*` targets in our `contrib/refresh_checksums.mk` invocations.

Reply:
Thank you for the insight Elliot! :)